cellmlmanip/model.py

```diff
@@ -689,17 +689,23 @@ def get_value(self, symbol):
 
     def get_initial_value(self, symbol):
         """Returns the initial value of the given symbol
+
         :param symbol: Sympy Dummy object of required symbol
         :return: float of initial value
         """
         return float(self.dummy_metadata[symbol].initial_value)
 
     def find_symbols_and_derivatives(self, expression):
-        """ Returns a set containing all symbols and derivatives referenced in an expression. """
-        symbols = set()
+        """ Returns a set containing all symbols and derivatives referenced in an expression or list of expressions.
```

Comment:
Normally I'd prefer always to accept just a list, but I see the method is recursive so not that simple. Still, having an argument called `expression` that could be `expressions` feels wrong!

Reply:
That's what it used to be before though

Reply:
Hopefully it's better now

Reply:
It only accepted lists before, but I changed that because I didn't see that functionality used. We can just change it back to what it was before!

Reply:
I'll stick the old code back in, but then once the chaste_cg PR is ready for review again we should take a long time to discuss which bits can go into cellmlmanip. To me this kind of access feels to "involved" for the user code, but I could be very wrong!

Reply:
> I'll stick the old code back in

Now implemented in #143

Reply:
Sure so what I'm doing thee is the following:
- create a list of dicts with for for each state var (for chaste output) such as name,  units etc.

However the chaste code has 2 different states of equations that are both preceded by **only** the state vars that are used in those equations (not all ll state vars are necessarily used in both).

Alternatives could be something like 
`state_vars_in([exp])` or `contains_var([expr], var)`

Reply:
From what I remember, pycml does the equivalent of `get_equations_for` to get all the equations, and then filters the full state var list accordingly using a set intersection. Not sure if the same approach would be easy with cellmlmanip as it's implemented currently?

Reply:
Moving to #144 !